### PR TITLE
feat: Introduce Label Style

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we introduce Label Style where we can use it and provide a certain style across all labels throughout the app.

Note: Base branch is set to `feature/CreateAColorLibraryAccordingToFigma` to keep changes to a minimum

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/Components/LabelStyle.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import UIKit
+
+public struct LabelStyle {
+    private(set) var backgroundColor: UIColor
+    private(set) var textColor: UIColor
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Components/LabelStyle.swift:24]: Any particular reason to have `private(set) var` instead of `let`?

Comment (agisilaos) [reply Wire-iOS/Sources/Components/LabelStyle.swift:24]: to make them not settable from outside the struct. But I guess let can work here as well.

Comment (johnxnguyen) [reply Wire-iOS/Sources/Components/LabelStyle.swift:24]: You might also want to just make them mutable with `var`, that might make it easier to configure and define variations based off other styles, considering we're using structs with value semantics. For instance, suppose I had one style, and wanted to define a new one which is based of the existing one:

```swift
let newStyle: LabelStyle = {
    var style = oldStyle
    style.textColor = .red
    return style
}()
```

Comment (agisilaos) [reply Wire-iOS/Sources/Components/LabelStyle.swift:24]: So we should change the `private(set) var backgroundColor: UIColor` to `var backgroundColor: UIColor` and the

Comment (johnxnguyen) [reply Wire-iOS/Sources/Components/LabelStyle.swift:24]: Personally I would do it, I see the style object as just a data class that holds various properties.

**Review by johnxnguyen — APPROVED** [review]

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -8032,6 +8035,7 @@
 				551CD27822BA288B00EE9AE5 /* LegalHoldDisclosureController.swift in Sources */,
 				876B94821D7D8C870063AE89 /* UIView+SlideInState.swift in Sources */,
 				5E35F736217F417200D3F4FE /* BurstTimestampTableViewCell.swift in Sources */,
+				E92D1B95285B773B00366AFC /* LabelStyle.swift in Sources */,
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Wire-iOS.xcodeproj/project.pbxproj:8038]: `Component styles` ?

**Pull request merged**